tests/test_mlkit.py

```diff
@@ -351,9 +481,180 @@ def test_model_source_validation_errors(self, model_source):
          ValueError)
     ])
     def test_gcs_tflite_source_validation_errors(self, uri, exc_type):
-        with pytest.raises(exc_type) as err:
+        with pytest.raises(exc_type) as excinfo:
             mlkit.TFLiteGCSModelSource(gcs_tflite_uri=uri)
-        check_error(err.value, exc_type)
+        check_error(excinfo, exc_type)
+
+    def test_wait_for_unlocked_not_locked(self):
+        model = mlkit.Model(display_name="not_locked")
+        model.wait_for_unlocked()
+
+    def test_wait_for_unlocked(self):
+        recorder = instrument_mlkit_service(status=200,
+                                            operations=True,
+                                            payload=OPERATION_DONE_PUBLISHED_RESPONSE)
+        model = mlkit.Model.from_dict(LOCKED_MODEL_JSON_1)
+        model.wait_for_unlocked()
+        assert model == FULL_MODEL_PUBLISHED
+        assert len(recorder) == 1
+        assert recorder[0].method == 'GET'
+        assert recorder[0].url == TestModel._op_url(PROJECT_ID, MODEL_ID_1)
+
+    def test_wait_for_unlocked_timeout(self):
+        instrument_mlkit_service(status=200, operations=True, payload=OPERATION_NOT_DONE_RESPONSE)
+        mlkit._MLKitService.POLL_BASE_WAIT_TIME_SECONDS = 5  # longer for timeout
+        model = mlkit.Model.from_dict(LOCKED_MODEL_JSON_1)
+        with pytest.raises(Exception) as excinfo:
+            model.wait_for_unlocked(max_time_seconds=3)
```

Comment:
Set this to a much smaller value please (e.g. 0.1). We want the unit tests to run very fast.

Alternatively, can we add a Sleeper type somewhere that can be modified for tests?

```
# mlkit.py

class Sleeper(object):
     def sleep(self, duration_seconds):
         time.sleep(duration_seconds)

_SLEEPER = Sleeper()


# Unit tests
class TestSleeper():
    def sleep(self, delay_seconds):
        pass

mlkit._SLEEPER = TestSleeper()
```

Reply:
All 104 tests run in under 1 second total. 
This max time in this test is designed to time out before it polls a single time because I set the POLL_BASE_WAIT_TIME_SECONDS to 5 seconds and the maximum wait time to 3 seconds, so when it calculates the after_sleep_time it will not poll at all (stop_time = x + 3, after_sleep_time = x + 5).